### PR TITLE
qa/tasks/salt: print debug info when ping fails due to no master

### PR DESCRIPTION
**Pull request by smithfarm**
In the CI we saw:

```
Running: "sudo sh -c 'salt \\* test.ping' 2>/dev/null || true"
The salt master could not be contacted. Is master running?
tasks.salt_manager:0 of 1 minions responded
Running: "sudo sh -c 'salt \\* test.ping' 2>/dev/null || true"
The salt master could not be contacted. Is master running?
tasks.salt_manager:0 of 1 minions responded
ERROR:teuthology.run_tasks:Saw exception from tasks.
Traceback (most recent call last):
  File "/home/ubuntu/teuthology/teuthology/run_tasks.py", line 95, in run_tasks
    run_one_task(taskname, stack, timer, ctx=ctx, config=config)
  File "/home/ubuntu/teuthology/teuthology/run_tasks.py", line 74, in run_one_task
    manager.__enter__()
  File "/home/ubuntu/teuthology/teuthology/task/__init__.py", line 121, in __enter__
    self.begin()
  File "/home/ubuntu/src/github.com_SUSE_ceph_ses6/qa/tasks/salt.py", line 191, in begin
    self.sm.ping_minions()
  File "/home/ubuntu/src/github.com_SUSE_ceph_ses6/qa/tasks/salt_manager.py", line 206, in ping_minions
    number_of_minions,
  File "/home/ubuntu/src/github.com_SUSE_ceph_ses6/qa/tasks/salt_manager.py", line 82, in __ping
    while proceed():
  File "/home/ubuntu/teuthology/teuthology/contextutil.py", line 132, in __call__
    raise MaxWhileTries(error_msg)
MaxWhileTries: 'sudo sh -c 'salt \* test.ping' 2>/dev/null || true'
reached maximum tries (50) after waiting for 750 seconds
```

But this was impossible to debug, hence this commit.

Signed-off-by: Nathan Cutler <ncutler@suse.com>

### CODE REVIEW

File: qa/tasks/salt_manager.py [hunk 1]
```diff
@@ -78,13 +78,22 @@ def __cat_file_remote(self, remote, filename=None):
...
-                if (expected == responded):
-                    return None
+        try:
+            def instances_of_str(search_str, output):
+                return len(re.findall(search_str, output))
+            with safe_while(sleep=15, tries=50,
+                            action=ping_cmd) as proceed:
+                while proceed():
+                    output = self.master_remote.sh(ping_cmd)
+                    no_master = instances_of_str('The salt master could not be contacted')
```
Comment (kshtsk) [inline qa/tasks/salt_manager.py]: You do not pass output to instances_of_str, that will make a exception?

File: qa/tasks/salt_manager.py [hunk 2]
```diff
@@ -78,13 +78,22 @@ def __cat_file_remote(self, remote, filename=None):
...
     def __ping(self, ping_cmd, expected):
-        with safe_while(sleep=15, tries=50,
-                        action=ping_cmd) as proceed:
-            while proceed():
-                responded = len(re.findall('  True', self.master_remote.sh(ping_cmd)))
-                log.info("{} of {} minions responded".format(responded, expected))
-                if (expected == responded):
-                    return None
+        try:
+            def instances_of_str(search_str, output):
```
Comment (kshtsk) [inline qa/tasks/salt_manager.py]: I think there is no advantage of using this shortcut.

Comment (kshtsk) [reply qa/tasks/salt_manager.py]: it is even longer than len/re combination:
```
>>> len('instances_of_str()')
18
>>> len('len(re.findall())')
17
```

Comment (smithfarm) [reply qa/tasks/salt_manager.py]: The advantage is code readability.

Comment (kshtsk) [reply qa/tasks/salt_manager.py]: In this particular case it even worse readability as for me.

File: qa/tasks/salt_manager.py (hunk 1 above)
Comment (smithfarm) [reply qa/tasks/salt_manager.py]: Good catch

**Comment by smithfarm** [comment]
Tested that, at least, it does not break the salt task.

**Pull request merged**